lib/rules/comma-dangle.js

```diff
@@ -12,7 +12,9 @@
 //------------------------------------------------------------------------------
 
 module.exports = function (context) {
-    var forbidDanglingComma = context.options[0] !== "always";
+    var allowDangle = context.options[0] || "never";
```

Comment:
Look at what happens when `context.options[0]` is `"foo"`:

* `forbidDanglingComma` is `true` (equivalent to `"never"`)
* `allowDangle` is `"foo"` (equivalent to `"always"` based on logic around line 38)